### PR TITLE
plugins/overseer: init

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves https://github.com/nix-community/nixvim/issues/1528

### CODE REVIEW

File: plugins/by-name/overseer-nvim/default.nix [hunk 1]
```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "overseer-nvim";
```
Comment (GaetanLepage) [inline plugins/by-name/overseer-nvim/default.nix:10]: Genuine question, why not `overseer` in this case ?

Comment (khaneliman) [reply plugins/by-name/overseer-nvim/default.nix:10]: Didn't know what the preference was for following repo name or lua setup name

Comment (GaetanLepage) [reply plugins/by-name/overseer-nvim/default.nix:10]: The convention we have been using until now mostly correspond to the lua name, i.e. plugin name without the `.nvim` suffix.

Comment (khaneliman) [reply plugins/by-name/overseer-nvim/default.nix:10]: Sounds good I'll change these init prs to do that

**Comment by GaetanLepage** [comment]
Could you also take a quick look at the previous draft #1759 just in case something interesting lies there ?

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by khaneliman** [comment]
> Could you also take a quick look at the previous draft #1759 just in case something interesting lies there ?

I didn't notice that PR but, it looks like they were going full verbose, I was just just aiming for the functionality specific options here in the declared options and leaving out the UI options. So, there were a couple extra keymap options I'll include since they did the heav lifting of defining them already.

**Comment by GaetanLepage** [comment]
> I didn't notice that PR but, it looks like they were going full verbose, I was just just aiming for the functionality specific options here in the declared options and leaving out the UI options. So, there were a couple extra keymap options I'll include since they did the heavy lifting of defining them already.

Ok good for me !

File: plugins/by-name/overseer/default.nix [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            n = {
+              "<CR>" = "Submit";
+              "<C-s>" = "Submit";
+              "q" = "Cancel";
+              "?" = "ShowHelp";
+            };
+          }
+          ''
+            Set keymap to false to remove default behavior
+            You can add custom keymaps here as well (anything vim.keymap.set accepts)
```
Comment (GaetanLepage) [inline plugins/by-name/overseer/default.nix:123]: Ensure to end all sentences with a period.
Like so, we can provide a consistent documentation style.

**Comment by khaneliman** [comment]
Also looks like the old keymap option might not be fully satisfactory.. Might need to update to follow https://github.com/nix-community/nixvim/issues/603

**Review by GaetanLepage — APPROVED** [review]

**Comment by khaneliman** [comment]
Just pushed a subtle change to untie the keymaps from our submodule and allow any input so we dont accidentally restrict a user from upstream keymap usage.

**Comment by khaneliman** [comment]
@mergifyio queue

**Pull request merged**